master/buildbot/test/unit/test_changes_gitpoller.py

```diff
@@ -285,6 +287,42 @@ def cb(_):
                 'master': '4423cdbcbb89c14e50dd5f4152415afd686c5241'
             })
 
+    def test_poll_GitError(self):
+        # Raised when git exits with status code 128. See issue 2468
+        self.expectCommands(
+            gpo.Expect('git', 'init', '--bare', 'gitpoller-work')
+            .exit(128),
+        )
+
+        @defer.inlineCallbacks
+        def run_command():
+            yield self.poller._dovccmd('init', ['--bare',
+                                       'gitpoller-work'])
+
+        d = self.assertFailure(run_command(), gitpoller.GitError)
+
+        d.addCallback(lambda _: self.assertAllCommandsRan())
+        return d
+
+    def test_poll_GitError_loop(self):
```

Comment:
Also here, need inline callbacks

I guess you also want to patch asyncSleep so that the test does not unneedly wait 1sec